R/SimulateSOI.R

```diff
@@ -0,0 +1,222 @@
+#' Self-Organised Instability Model Community Simulation
+#'
+#' Generate time-series with The Self-Organised Instability (SOI) model.
+#' Implements a K-leap method for accelerating stochastic simulation.
+#'
+#' @param N Integer: number of species
+#' @param I Integer: community size, number of available sites (individuals)
+#' @param A a interaction matrix of dimension NxN
+#' @param com a vector inditicating initial community abundances.
+#' If NULL (default), based on migration rates.
+#' @param tend Integer: number of timepoints to be returned in the time series
+#' (nr of generations)
+#' @param k Integer: the number of transition events that are allowed to take
+#' place during one leap. By default set to 5. Higher values reduce runtime,
+#' but also accuracy of the simulation.
+#' @param norm logical to indicate whether the time series should be returned
+#' with the abundances as proportions (norm = TRUE)
+#' or the raw counts (norm = FALSE, default)
+#' @return \linkS4class{SummarizedExperiment} object with abundance matrix
+#' consisting of species abundance as rows and time points as columns
+#' @examples simulateSOI(N = 10, I = 1000, A = powerlawA(n = 10, alpha = 1.2),
+#'                     k=5, com = NULL, tend = 150, norm = TRUE)
+#' @export
+
+setGeneric("simulateSOI",signature = "N",
+           function(N, I, A, k, com, tend, norm = FALSE)
+             standardGeneric("simulateSOI"))
+
+#' @export
+
+powerlawA <- function(
+  n, # number of species
+  alpha, # power-law distribution parameter
+  stdev = 1, # sd normal distribution
+  s = 0.1 # scaling parameter, default: 0.1/max(A)
+){
+  # Nominal Interspecific Interaction matrix N
+  N <- matrix(
+    data = rnorm(n^2, mean = 0, sd = stdev),
+    nrow = n,
+    ncol = n
+  )
+  # power law sample
+  pl <- rplcon(n = n, xmin = 1, alpha = alpha)
+  # Interaction strength heterogeneity H
+  H <- sapply(seq_len(n), FUN = function(i){
+    1 + ((pl[i]-min(pl))/(max(pl)-min(pl)))
+  })
+  H <- diag(H)
+  # Adjacency matrix G of power-law out-degree digraph ecological network
+  d <- 0.1*n
+  h <- sapply(seq_len(n), FUN = function(i){
+    min(
+      ceiling(d*pl[i]/mean(pl)),
+      n
+    )
+  })
+  G <- matrix(0, nrow = n, ncol = n)
+  for(i in seq_len(n)){
+    index <- sample(x = seq_len(n), size = h[i])
+    G[index, i] <- 1
+  }
```

Comment:
This does not follow the 4-space indentation required by Bioconductor. Is there no warning about this in BiocCheck()?

Reply:
Same for rest of the code

Reply:
Sorry, i copy pasted to new file just before uploading, it must be because of that. Now it is corrected and checked